source/SVInt.cpp

```diff
@@ -1690,4 +1699,107 @@ bool exactlyEqual(const SVInt& lhs, const SVInt &rhs) {
     return memcmp(lhs.pVal, rhs.pVal, lhs.getNumWords() * SVInt::WORD_SIZE) == 0;
 }
 
+bool wildcardEqual(const SVInt& lhs, const SVInt& rhs) {
+    // if no unknown flags, do normal comparison
+    if (!lhs.unknownFlag && !rhs.unknownFlag)
+        return (bool)(lhs == rhs);
+
+    // handle sign extension if necessary
+    if (lhs.bitWidth != rhs.bitWidth) {
+        bool bothSigned = lhs.signFlag && rhs.signFlag;
+        if (lhs.bitWidth < rhs.bitWidth)
+            return wildcardEqual(extend(lhs, rhs.bitWidth, bothSigned), rhs);
+        else
+            return wildcardEqual(lhs, extend(rhs, lhs.bitWidth, bothSigned));
+    }
+
+    size_t words = SVInt::getNumWords(lhs.bitWidth, false);
+    // Do word-for-word wildcard comparison, being careful that one argument
+    // might nothave unknwon flags
+    for (size_t i = 0; i < words; ++i) {
+        // bitmask of bits that actually matter in the comparison
+        uint64_t mask = ~((lhs.unknownFlag ? lhs.pVal[i + words] : 0) |
```

Comment:
X and Z are only wildcards in the rhs; in the lhs side they either match a wildcard, or they cause the result to be X (so this function can't just return a bool).